// @oracle/bots-node-sdk/typings/lib2/messagev2/field/field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Margin = exports.FontWeight = exports.FontSize = exports.Field = void 0;
const internal_1 = require("../internal");
/**
 * Represents an abstract field.
 * @extends ChannelCustomizable
 */
class Field extends internal_1.ChannelCustomizable {
    // properties below will be added in 23.08
    // private marginTop: Margin;
    // private labelFontSize: FontSize;
    // private labelFontWeight: FontWeight;
    /**
     * Gets the label of the field.
     * @returns {string} The label of the field.
     */
    getLabel() {
        return this.label;
    }
    /**
     * Sets the label of the field.
     * @param {string} label The label of the field.
     * @returns This Field instance.
     */
    setLabel(label) {
        this.label = label;
        return this;
    }
    /**
     * Gets the ID of the field.
     * @returns {string} The ID of the field.
     */
    getId() {
        return this.id;
    }
    /**
     * Sets the ID of the field.
     * @param {string} id The ID of the field.
     * @returns This Field instance.
     */
    setId(id) {
        this.id = id;
        return this;
    }
}
exports.Field = Field;
/**
 * Represents the font size options for a field.
 */
var FontSize;
(function (FontSize) {
    FontSize["small"] = "small";
    FontSize["medium"] = "medium";
    FontSize["large"] = "large";
})(FontSize = exports.FontSize || (exports.FontSize = {}));
/**
 * Represents the font weight options for a field.
 */
var FontWeight;
(function (FontWeight) {
    FontWeight["light"] = "light";
    FontWeight["medium"] = "medium";
    FontWeight["bold"] = "bold";
})(FontWeight = exports.FontWeight || (exports.FontWeight = {}));
/**
 * Represents the margin options for a field.
 */
var Margin;
(function (Margin) {
    Margin["none"] = "none";
    Margin["medium"] = "medium";
    Margin["large"] = "large";
})(Margin = exports.Margin || (exports.Margin = {}));
